about-pses/example-pse.md

```diff
@@ -44,17 +44,36 @@ List three or more questions whose answers would clarify the problem statement.
+2. Should a user automatically lose if they have invalid input? I'll assume the input cannot be invalid. My function must be part of a larger system (something must be calling it!) and that system should handle input validation. This way, I can focus on the core logic of the scoring and not worry about input validation.
+3. What should happen if both users have invalid input? While both players providing invalid input could be considered a tie, I don't want the function to act as though everything was fine if an input was invalid, so I'll return `None` if either player has invalid input. This way, the caller can detect that something went wrong and decide how to handle it. A drawback of this approach is that the caller can't tell _which_ input was invalid, only that one of them was.
+4. Does capitalization matter? I'll assume that the input is case-insensitive, so I'll convert the input to lowercase before comparing it to the valid moves. This way, the caller can provide input in any case they like, and the function will still work. This could be helpful if this code was incorporated into a mobile platform where the device might auto-capitalize the first letter of a word.
+5. Does extra whitespace / punctuation matter? I'll assume that my function is part of a larger game (something must be calling my function, right?) and that the caller will handle input validation. I will implement my logic looking only for the exact strings "rock", "paper", and "scissors". This way, the caller can handle any input validation they need to do, and I can focus on the core logic of the scoring.
+6. I'm familiar with a variant of rock, paper, scissors that includes additional throws (lizard and Spock). Should I keep my approach focused on the basic rules, or should I use an implementation that could be extended to game variants? While I could write a helper function that accepts a data structure that encodes the valid moves and what beats what, I'll assume that I should focus on the basic rules for now. This way, I can get a working solution faster, and if I need to extend the game later, I can refactor my code to be more flexible.
+7. When friends use rock, paper, scissors to make decisions, there are often more than two players. Should I keep my approach focused on the basic rules, or should I use an implementation that could be extended to support multiple players simultaneously? While I could write a helper function that accepts a list of moves made by all players, the multiplayer version introduces a few complications into resolving a round that I'd like to avoid for now. I'll assume that I should focus on the basic rules for now. This way, I can get a working solution faster, and if I need to extend the game later, I can refactor my code to be more flexible.
```

Comment:
These show up as dense chunks of text that my eyes struggle a little with. How would you feel about bolding the question and moving the answer to a sub-bullet?
```suggestion
1. **How should the function handle invalid user input (i.e. player_1 = "lizard")?** 
    * The prompt doesn't specify how to handle invalid input, so I'll return `None`, which would allow the caller of my function to detect that there was an input I wasn't expecting. The caller could then decide how to handle that situation. This is also a convenient decision because, I'll only need to look for valid winning combinations, and if I don't find one, I can return `None`.
2. **Should a user automatically lose if they have invalid input?** 
    * I'll assume the input cannot be invalid. My function must be part of a larger system (something must be calling it!) and that system should handle input validation. This way, I can focus on the core logic of the scoring and not worry about input validation.
3. **What should happen if both users have invalid input?** 
    * While both players providing invalid input could be considered a tie, I don't want the function to act as though everything was fine if an input was invalid, so I'll return `None` if either player has invalid input. This way, the caller can detect that something went wrong and decide how to handle it. A drawback of this approach is that the caller can't tell _which_ input was invalid, only that one of them was.
4. **Does capitalization matter?** 
    * I'll assume that the input is case-insensitive, so I'll convert the input to lowercase before comparing it to the valid moves. This way, the caller can provide input in any case they like, and the function will still work. This could be helpful if this code was incorporated into a mobile platform where the device might auto-capitalize the first letter of a word.
5. **Does extra whitespace / punctuation matter?** 
    * I'll assume that my function is part of a larger game (something must be calling my function, right?) and that the caller will handle input validation. I will implement my logic looking only for the exact strings "rock", "paper", and "scissors". This way, the caller can handle any input validation they need to do, and I can focus on the core logic of the scoring.
6. **I'm familiar with a variant of rock, paper, scissors that includes additional throws (lizard and Spock). Should I keep my approach focused on the basic rules, or should I use an implementation that could be extended to game variants?**
    * While I could write a helper function that accepts a data structure that encodes the valid moves and what beats what, I'll assume that I should focus on the basic rules for now. This way, I can get a working solution faster, and if I need to extend the game later, I can refactor my code to be more flexible.
7. **When friends use rock, paper, scissors to make decisions, there are often more than two players. Should I keep my approach focused on the basic rules, or should I use an implementation that could be extended to support multiple players simultaneously?** 
    * While I could write a helper function that accepts a list of moves made by all players, the multiplayer version introduces a few complications into resolving a round that I'd like to avoid for now. I'll assume that I should focus on the basic rules for now. This way, I can get a working solution faster, and if I need to extend the game later, I can refactor my code to be more flexible.
```

Reply:
This does look a lot better. Thank you!